packages/react/src/primitives/SelectField/SelectField.tsx

```diff
@@ -10,6 +10,27 @@ import { SelectFieldProps, Primitive } from '../types';
 import { splitPrimitiveProps } from '../shared/styleUtils';
 import { useStableId } from '../shared/utils';
 
+const selectFieldChildren = (
+  children?: React.ReactNode,
+  options?: string[]
+) => {
+  if (options?.length) {
+    if (children) {
+      console.warn(
+        'Amplify UI: <SelectField> component defaults to rendering children over `options`. When using the `options` prop, do not also pass children.'
+      );
+    } else {
```

Comment:
What do you think of doing an early return for the `children`, since `children` takes precedence? 

```jsx
  if (children) {
    if (options?.length) {
          console.warn(
             'Amplify UI: <SelectField> component  defaults to rendering children over `options`. When using the `options` prop, omit children.'
          );
    }
    return children;
  }
  
  return options.map((option, index) => (
        <option label={option} value={option} key={index}>
          {option}
        </option>
  ));



```

IMO, else statements are usually a code smell that the logic should be simplified (often using an early return).

Reply:
Good point. The reason I wrote it this way is because I wanted to handle the case where there are neither `children` nor `options`.